fhir-server/src/main/java/com/ibm/fhir/server/resources/FHIRResource.java

```diff
@@ -1290,6 +1299,12 @@ private FHIRRestOperationResponse doPatchOrUpdate(String type, String id, FHIRPa
                     String msg = "The 'id' parameter is required for an update/pach operation.";
                     throw buildRestException(msg, Status.BAD_REQUEST, IssueType.ValueSet.REQUIRED);
                 }
+                
+                // Make sure the resource has an 'id' attribute.
+                if (newResource.getId() == null) {
```

Comment:
out of curiosity - at this point?  why would `newResource.getId() == null`?

Reply:
yeah, it can happen if there is no resource id in the request body. according to the spec, in this case, there should always be a resource id in the request body, otherwise, we should 400.

Reply:
OK. 

Should ValueSet.Required be set on 1306?  like line 1300?

Reply:
I think Invalid should also be fine, because it means resource without an id in the request body is invalid content...

Reply:
Right - just double checking for consistency.

Reply:
I think invalid should also be fine, it means the resource without id in the request body is a invalid contents.